chemprop/uncertainty/predictor.py

```diff
@@ -291,10 +310,56 @@ def __call__(
 
 @UncertaintyPredictorRegistry.register("multiclass-dirichlet")
 class MulticlassDirichletPredictor(UncertaintyPredictor):
+    """
+    A :class:`MulticlassDirichletPredictor` predicts an amount of 'evidence' for each class as
+    described in [sensoy2018]_. The class probabilities and the uncertainty are calculated based on
+    the evidence.
+
+    .. math::
+        S = \sum_{i=1}^K \alpha_i
+        p_i = \alpha_i / S
+        u = K / S
+
+    where :math:`K` is the number of classes, :math:`\alpha_i` is the evidence for class :math:`i`,
+    :math:`p_i` is the probability of class :math:`i`, and :math:`u` is the uncertainty.
+
+    References
+    ----------
+    .. [sensoy2018] Sensoy, M.; Kaplan, L.; Kandemir, M. "Evidential deep learning to quantify
+        classification uncertainty." NeurIPS, 2018, 31. https://doi.org/10.48550/arXiv.1806.01768
+    """
+
     def __call__(
         self, dataloader: DataLoader, models: Iterable[MPNN], trainer: pl.Trainer
     ) -> tuple[Tensor, Tensor]:
-        return
+        preds = []
+        uncs = []
+        for model in models:
+            self._setup_model(model)
+            output = torch.concat(trainer.predict(model, dataloader), 0)
+            self._restore_model(model)
+            preds.append(output[..., :-1])
+            uncs.append(output[..., -1])
+            preds = torch.stack(preds, 0)
+            uncs = torch.stack(uncs, 0).unsqueeze(-1).repeat(1, 1, 1, preds.shape[-1])
```

Comment:
```suggestion
        preds = torch.stack(preds, 0)
        uncs = torch.stack(uncs, 0)
```
Should we change the docstring to reflect the shape of multiclass classification as `m' n x t` instead of `m' n x t x c`?